src/CpAmm.ts

```diff
@@ -424,25 +425,49 @@ export class CpAmm {
 
     const instructions: TransactionInstruction[] = [];
 
-    // 1. claim position fee
-    const claimPositionFeeInstruction =
-      await this.buildClaimPositionFeeInstruction({
-        owner,
-        poolAuthority: this.poolAuthority,
-        pool,
-        position,
-        positionNftAccount,
-        tokenAAccount,
-        tokenBAccount,
-        tokenAVault,
-        tokenBVault,
-        tokenAMint,
-        tokenBMint,
-        tokenAProgram,
-        tokenBProgram,
-      });
+    // 1. claim position fee if it available
+    const unclaimedReward = getUnClaimReward(poolState, positionState);
+    const hasUnclaimedFee =
+      unclaimedReward.feeTokenA.gt(new BN(0)) ||
+      unclaimedReward.feeTokenB.gt(new BN(0));
 
-    instructions.push(claimPositionFeeInstruction);
+    if (hasUnclaimedFee) {
+      const claimPositionFeeInstruction =
+        await this.buildClaimPositionFeeInstruction({
+          owner,
+          poolAuthority: this.poolAuthority,
+          pool,
+          position,
+          positionNftAccount,
+          tokenAAccount,
+          tokenBAccount,
+          tokenAVault,
+          tokenBVault,
+          tokenAMint,
+          tokenBMint,
+          tokenAProgram,
+          tokenBProgram,
+        });
+
+      instructions.push(claimPositionFeeInstruction);
+    }
+
+    if (positionState.rewardInfos.length > 0) {
+      for (let index = 0; index < positionState.rewardInfos.length; index++) {
+        if (positionState.rewardInfos[index].rewardPendings.isZero()) continue;
+
+        const claimRewardTx = await this.claimReward({
+          user: owner,
+          position,
+          positionNftAccount,
+          rewardIndex: index,
+          poolState,
+          positionState,
+        });
+
+        instructions.push(...claimRewardTx.instructions);
```

Comment:
no idea if this pr is still relevant, but `claimReward` appends pre and post ixs so there would be dupe ixs if `positionState.rewardInfos.length > 1`